Release RC2 (#271)
JoshOrndorff/recipes: text/3-entrees/weights.md

# Computational Resources and Weights

_[`pallets/weights`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/weights)_

Any computational resources used by a transaction must be accounted for so that appropriate fees can
be applied, and it is a pallet author's job to ensure that this accounting happens. Substrate
provides a mechanism known as transaction weighting to quantify the resources consumed while
executing a transaction.

> _Indeed, mispriced EVM operations have shown how operations that underestimate cost can open

Reviewer: ```suggestion
_Indeed, mispriced EVM operations have shown how operations that underestimate cost can provide
```

		}
```

For more complex transactions, custom weight calculations can be performed that consider the
parameters passed to the call. This snippet shows a weighting struct that weighs transactions where

> the first parameter is a `boo`l. If the first parameter is `true`, then the weight is linear in the

Reviewer: ```suggestion
the first parameter is a `bool`. If the first parameter is `true`, then the weight is linear in the
```

}
```

In addition to the
[`WeightData` Trait](https://substrate.dev/rustdocs/v2.0.0-rc2/frame_support/weights/trait.WeighData.html), shown

> above, types that are used to calculate transaction weights, must also implement

Reviewer: ```suggestion
above, types that are used to calculate transaction weights must also implement
```
